docs/assembly.rst

```diff
@@ -1,66 +1,31 @@
 especially when you are enhancing the language by writing libraries.
 
-As the EVM is a stack machine, it is often hard to address the correct stack slot
-and provide arguments to opcodes at the correct point on the stack. Solidity's inline
-assembly helps you do this, and with other issues that arise when writing manual assembly.
+The language used for inline assembly in Solidity is called `Yul <yul>`_
+and it is documented in its own section. This section will only cover
+how the inline assembly code can interface with the surrounding Solidity code.
 
-For inline assembly, the stack is actually not visible at all, but if you look
-closer, there is always a very direct translation from inline assembly to
-the stack based EVM opcode stream.
-
-Inline assembly has the following features:
-
-* functional-style opcodes: ``mul(1, add(2, 3))``
-* assembly-local variables: ``let x := add(2, 3)  let y := mload(0x40)  x := add(x, y)``
-* access to external variables: ``function f(uint x) public { assembly { x := sub(x, 1) } }``
-* loops: ``for { let i := 0 } lt(i, x) { i := add(i, 1) } { y := mul(2, y) }``
-* if statements: ``if slt(x, 0) { x := sub(0, x) }``
-* switch statements: ``switch x case 0 { y := mul(x, 2) } default { y := 0 }``
-* function calls: ``function f(x) -> y { switch x case 0 { y := 1 } default { y := mul(x, f(sub(x, 1))) }   }``
 
 .. warning::
     Inline assembly is a way to access the Ethereum Virtual Machine
     at a low level. This bypasses several important safety
     features and checks of Solidity. You should only use it for
     tasks that need it, and only if you are confident with using it.
 
-Syntax
-------
-
-Assembly parses comments, literals and identifiers in the same way as Solidity, so you can use the
-usual ``//`` and ``/* */`` comments. There is one exception: Identifiers in inline assembly can contain
-``.``. Inline assembly is marked by ``assembly { ... }`` and inside
-these curly braces, you can use the following (see the later sections for more details):
 
- - literals, i.e. ``0x123``, ``42`` or ``"abc"`` (strings up to 32 characters)
- - opcodes in functional style, e.g. ``add(1, mload(0))``
- - variable declarations, e.g. ``let x := 7``, ``let x := add(y, 3)`` or ``let x`` (initial value of 0 is assigned)
- - identifiers (assembly-local variables and externals if used as inline assembly), e.g. ``add(3, x)``, ``sstore(x_slot, 2)``
- - assignments, e.g. ``x := add(y, 3)``
- - blocks where local variables are scoped inside, e.g. ``{ let x := 3 { let y := add(x, 1) } }``
+Inline assembly is marked by ``assembly { ... }``, where the code inside
+the curly braces is `Yul <yul>`_ code.
 
-Inline assembly manages local variables and control-flow. Because of that,
-opcodes that interfere with these features are not available. This includes
-the ``dup`` and ``swap`` instructions as well as ``jump`` instructions and labels.
+This Yul code has access to local Solidity variables as explained below.
```

Comment:
```suggestion
This Yul code has access to Solidity variables in scope of the inline assembly block as explained below.
```
It's not only locals, is it?

Reply:
I hoped this was obvious.

Reply:
Yeah, maybe... but then why say "local" at all?